### PR TITLE
wasm: remove ASYNCIFY to reduce binary size

### PR DESCRIPTION
**Pull request by tinyjin**
The binary size recently increased due to the ASYNCIFY option, which was required for WebGPU initialization. 

To prevent unnecessary binary size growth, ThorVG will no longer depend on asynchronous processes such as `emscripten_sleep` (ASYNCIFY or JSPI).

As a result, the combined WASM binary size has been significantly reduced to less than 1MB.

Size comparison: 1559KB → 998KB (-36%)

CleanShot 2024-12-09 at 19 59 51@2x

CleanShot 2024-12-09 at 19 55 19@2x

### CODE REVIEW

**Review by hermet — COMMENTED** [review]
@tinyjin What is the user supposed to do when init() returns false?

**Comment by hermet** [comment]
https://github.com/thorvg/thorvg.web/pull/80

**Comment by tinyjin** [comment]
> @tinyjin What is the user supposed to do when init() returns false?

Call `wait(100)` function in JS side instead of `emscripten_sleep(100)`

**Comment by hermet** [comment]
> > @tinyjin What is the user supposed to do when init() returns false?
> 
> Call `wait(100)` function in JS side instead of `emscripten_sleep(100)`
> 
> `false` means initialization is not yet ready -> should wait until it gets done.

@tinyjin User should use an infinite loop for initialization. How to handle if adapter or device acquisition fails?

File: src/bindings/wasm/tvgWasmLottieAnimation.cpp [hunk 1]
```diff
@@ -41,32 +41,44 @@ static const char* NoError = "None";
     static WGPUInstance instance{};
     static WGPUAdapter adapter{};
     static WGPUDevice device{};
+    static bool adapterRequested = false;
+    static bool deviceRequested = false;
 #endif
 
-void init()
+bool init()
```
Comment (hermet) [inline src/bindings/wasm/tvgWasmLottieAnimation.cpp:48]: int // 0: success, 1: fail, 2: try again

File: src/bindings/wasm/tvgWasmLottieAnimation.cpp [hunk 2]
```diff
@@ -41,32 +41,60 @@ static const char* NoError = "None";
...
 
-void init()
+// 0: success, 1: fail, 2: wait for async request
+int init()
 {
 #ifdef THORVG_WG_RASTER_SUPPORT
     //Init WebGPU
     if (!instance) instance = wgpuCreateInstance(nullptr);
 
+    if (initializationFailed) return 1;
```
Comment (hermet) [inline src/bindings/wasm/tvgWasmLottieAnimation.cpp:56]: this should be at 53 line?

File: src/bindings/wasm/tvgWasmLottieAnimation.cpp [hunk 3]
```diff
@@ -41,32 +41,60 @@ static const char* NoError = "None";
...
+
     // request adapter
     if (!adapter) {
+        if (adapterRequested) return 2;
+
         const WGPURequestAdapterOptions requestAdapterOptions { .nextInChain = nullptr, .powerPreference = WGPUPowerPreference_HighPerformance, .forceFallbackAdapter = false };
-        auto onAdapterRequestEnded = [](WGPURequestAdapterStatus status, WGPUAdapter adapter, char const * message, void * pUserData) { *((WGPUAdapter*)pUserData) = adapter; };
```
Comment (hermet) [inline src/bindings/wasm/tvgWasmLottieAnimation.cpp:55]: coding convention:
-> `char const* message, void* pUserData`

File: src/bindings/wasm/tvgWasmLottieAnimation.cpp [hunk 4]
```diff
@@ -41,32 +41,60 @@ static const char* NoError = "None";
...
+        auto onAdapterRequestEnded = [](WGPURequestAdapterStatus status, WGPUAdapter adapter, char const * message, void * pUserData) {
```
Comment (hermet) [inline src/bindings/wasm/tvgWasmLottieAnimation.cpp:63]: coding convention:
-> `char const* message, void* pUserData`

**Review by hermet — APPROVED** [review]
please double-check the comments and push yourself. thanks.

**Pull request merged**